### PR TITLE
Workflow Status Monitoring Schema

### PR DESCRIPTION
**Pull request by AntoineDao**
This PR addresses the workflow status tracking primitives issue that will be used by plugins to provide a uniform debugging schema.

### CODE REVIEW

File: queenbee/schema/status.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        ...,
+        description='The inputs used by this task'
+    )
+
+    outputs: Arguments = Schema(
+        ...,
+        description='The outputs produced by this task'
+    )
+
+    boundary_id: str = Schema(
```
[the remark this answers is not in the document]
Comment (AntoineDao) [reply queenbee/schema/status.py]: This is what I was worried about pulling stuff straight from Argo! The boundary_id is the upper scope object containing whatever this task is. So for example:

```
Workflow A
|_ DAG A
| |_ TASK A
| |_ TASK B
|_DAG B
  |_ TASK C
  |_ TASK D
```

DAGs A and B have a boundary ID of Workflow A and Task D has a boundary ID of DAG B. Does that make sense?

[the remark this answers is not in the document]
Comment (AntoineDao) [reply queenbee/schema/status.py]: PS: Happy to make it optional.

File: queenbee/schema/status.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    name: str = Schema(
+        ...,
+        description='A human readable name for the task. Usually defined by the \
+            DAG task name but can be extended if the task is part of a loop for example. \
+            This name is unique within the boundary of the DAG/Workflow that generated it.'
+    )
+
+    type: str = Schema(
+        ...,
+        description='The type of task this status is for. Can be "Function", "DAG", "Workflow" or "Loop"'
```
[the remark this answers is not in the document]
Comment (AntoineDao) [reply queenbee/schema/status.py]: We can yeah.

File: queenbee/schema/status.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mostaphaRoudsari) [reply queenbee/schema/status.py]: It totally does but I think making it optional will make it easier for one to get started using this object.

[the remark this answers is not in the document]
Comment (AntoineDao) [reply queenbee/schema/status.py]: Yep will do.

**Pull request merged**